common-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java

```diff
@@ -0,0 +1,83 @@
+package org.pmiops.workbench.cdr.dao;
+
+import java.util.ArrayList;
+import java.util.List;
+import javax.persistence.EntityManager;
+import javax.persistence.criteria.Expression;
+import javax.persistence.criteria.Predicate;
+import org.pmiops.workbench.cdr.model.Concept;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Slice;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.domain.Sort.Direction;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ConceptService {
+
+  private final EntityManager entityManager;
+
+  public ConceptService(EntityManager entityManager) {
+    this.entityManager = entityManager;
+  }
+
+  public static final String STANDARD_CONCEPT_CODE = "S";
+
+  public Slice<Concept> searchConcepts(String query,
+      Boolean standardConcept, String vocabularyId, String domainId,
```

Comment:
I would make this a `boolean` and push the default-picking logic up to the API level

Reply:
So actually, I'm treating this as tri-state: standardConcept being null means both standard and non-standard concepts are returned, whereas true or false return only standard or non-standard concepts respectively.

If you prefer we could not bother with the "only non-standard concepts" or have an enum I guess.

Reply:
In general I don't like using null/undefined as a sentinel value for a boolean, so would prefer to collapse these or switch to an enum. On that decision, the usual API rule is: if a field has two possible values now, but could have more later, use an enum over a boolean.

Reply:
OK. I changed this to be an enum, as well as domain (which is now being mapped from a set of domain enum values to underlying domain IDs in ConceptsController.)

Note that there's some duplication of the StandardConceptFilter enums between the client and service... at the moment we don't do any swagger-codegen in common-api so this is necessary. At some point we may move the API code into common-api as well, and then we can get rid of this.